integrated_channels/integrated_channel/models.py

```diff
@@ -62,3 +77,55 @@ class EnterpriseCustomerPluginConfiguration(TimeStampedModel):
 
     class Meta:
         abstract = True
+
+    def transmit_learner_data(self):
+        """
+        Collect and transmit learner data for the enterprise_customer.
+        """
+        for learner_data in self.collect_learner_data():
+            LOGGER.info(learner_data)
+
+            # TODO ENT-221 will handle transmitting the learner data to the channel
+
+    def collect_learner_data(self):
+        """
+        Collect learner data for the enterprise_customer where data sharing consent is granted.
+
+        Yields a dict for each enrollment, containing:
+
+        * enterprise_enrollment: EnterpriseCourseEnrollment
+        * certificate: GeneratedCertificate for the user+course, None if not found.
+          "Course completion" occurs when course certificates are issued.
+        """
+
+        # Fetch the consenting enrollment data, including the enterprise_customer_user
+        enrollment_queryset = EnterpriseCourseEnrollment.objects.select_related(
+            'enterprise_customer_user'
+        ).filter(
+            consent_granted=True,
+            enterprise_customer_user__enterprise_customer=self.enterprise_customer,
+        )
+        for enterprise_enrollment in enrollment_queryset:
+
+            if CourseKey is None:
```

Comment:
Looks like you might also need to protect against `GeneratedCertificate` being `None` from the import